### PR TITLE
fix: Updated metal-go client for sub-commands gateways

### PR DESCRIPTION
**Pull request by codinja1188**
Breakout from https://github.com/equinix/metal-cli/pull/270

What this PR does / why we need it:

This PR replaces packngo with metal-go for all interactions with the Equinix Metal API specifically for gateways sub commands

DISCUSSION POINTS:

### CODE REVIEW

File: internal/pagination/pager.go [hunk 1]
```diff
@@ -145,3 +145,22 @@ func GetProjectDevices(s metal.ApiFindProjectDevicesRequest) ([]metal.Device, er
 		return devices, nil
 	}
 }
+
+func GetMetalGatewaysByProject(s metal.ApiFindMetalGatewaysByProjectRequest) ([]metal.MetalGatewayListMetalGatewaysInner, error) {
```
Comment (displague) [inline internal/pagination/pager.go]: We can get pagination for free now from metal-go

**Comment by cprivitere** [comment]
Needs update for new pagination and rebase.

File: internal/pagination/pager.go (hunk 1 above)
Comment (codinja1188) [reply internal/pagination/pager.go]: Updated code

**Comment by aayushrangwala** [comment]
@vasubabu Please decline this PR in the favour of https://github.com/equinix/metal-cli/pull/376

**Pull request closed without merging**